nni/mutable/utils.py

```diff
@@ -0,0 +1,226 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from __future__ import annotations
+
+__all__ = [
+    'uid', 'reset_uid',
+    'NoContextError', 'ContextStack', 'get_current_context',
+    'LabelNamespace', 'auto_label',
+]
+
+import logging
+from collections import defaultdict
+from typing import Any
+
+_last_uid = defaultdict(int)
+
+_logger = logging.getLogger(__name__)
+
+
+def uid(namespace: str = 'default') -> int:
+    """Global counter for unique id. Not thread-safe."""
+    _last_uid[namespace] += 1
+    return _last_uid[namespace]
+
+
+def reset_uid(namespace: str = 'default') -> None:
+    """Reset counter for a specific namespace."""
+    _last_uid[namespace] = 0
+
+
+class NoContextError(Exception):
+    """Exception raised when context is missing."""
+    pass
+
+
+class ContextStack:
```

Comment:
who will use this class? is it possible someone using it like:

```
with ContextStack(namespace, value):
    ContextStack.push(namespace, value2)
    ...
...
ContextStack.pop(namespace)
```

It is ok if only `LabelNamespace` using it.

Reply:
Haven't considered it carefully.

Right now it's mainly for internal usages. It's used in LabelNamespace and ModelSpace, to provide a global stack to access current active xxx. I'm not sure whether it's worth it to further polish it.